proto/src/main/java/com/google/gson/protobuf/LenientBase64Coder.java

```diff
@@ -0,0 +1,192 @@
+// Copyright 2003-2010 Christian d'Heureuse, Inventec Informatik AG, Zurich, Switzerland
+// www.source-code.biz, www.inventec.ch/chdh
+//
+// This module is multi-licensed and may be used under the terms
+// of any of the following licenses:
+//
+//  EPL, Eclipse Public License, V1.0 or later, http://www.eclipse.org/legal
+//  LGPL, GNU Lesser General Public License, V2.1 or later, http://www.gnu.org/licenses/lgpl.html
+//  GPL, GNU General Public License, V2 or later, http://www.gnu.org/licenses/gpl.html
+//  AL, Apache License, V2.0 or later, http://www.apache.org/licenses
+//  BSD, BSD License, http://www.opensource.org/licenses/bsd-license.php
+//  MIT, MIT License, http://www.opensource.org/licenses/MIT
+//
+// Please contact the author if you need another license.
+// This module is provided "as is", without warranties of any kind.
+
+package com.google.gson.protobuf;
+
+/**
+ * A Base64 encoder/decoder.
+ *
+ * <p> This class is used to encode and decode data in Base64 format as described in RFC 1521.
+ *
+ * <p> Project home page: <a href="http://www.source-code.biz/base64coder/java/">www.source-code.biz/base64coder/java</a><br>
+ * Author: Christian d'Heureuse, Inventec Informatik AG, Zurich, Switzerland<br> Multi-licensed: EPL
+ * / LGPL / GPL / AL / BSD / MIT.
+ *
+ * Modified to accept URL encoding, some apis removed from the original.
+ */
+public class LenientBase64Coder {
```

Comment:
Are you particularly attached to this base64 coded? I’m more experienced with the one in okio & it has a few little benefits:
https://github.com/square/okio/blob/master/okio/src/main/java/okio/Base64.java 

It ignores whitespace.
It doesn't create an extra copy of the char array, instead operating on the `String` directly
It can decode base64url
It isn't broken for inputs longer than `(Integer.MAX_VALUE / 3)`